app/controllers/orderitems_controller.rb

```diff
@@ -0,0 +1,45 @@
+class OrderitemsController < ApplicationController
+
+  def index
+    redirect_to order_path(current_order.id)
+  end
+
+  def create
+    @product = Product.find(params[:product_id])
+    @orderitem = current_order.orderitems.create(product: @product, quantity: params[:quantity], user_id: @product.user_id)
+    @orderitem.update(price: @orderitem.total_price)
+    redirect_to edit_order_path(current_order.id)
+  end
```

Comment:
I don't think your variables need to be instance variables (with the @ in front of them), since they're only used locally within the controller.